docs-en/chapter_hashing/hash_collision.md

```diff
@@ -1,45 +1,45 @@
 
-Hash collisions can lead to incorrect query results, severely affecting the usability of hash tables. To solve this problem, we expand the hash table whenever a hash collision occurs, until the collision is resolved. This method is simple and effective but inefficient due to the extensive data transfer and hash value computation involved in resizing the hash table. To improve efficiency, we can adopt the following strategies:
+Hash collisions can lead to incorrect query results, severely impacting the usability of the hash table. To address this issue, whenever a hash collision occurs, we perform hash table resizing until the collision disappears. This method is simple, straightforward, and effective, but its efficiency is too low because hash table resizing requires a large amount of data movement and hash value calculation. To improve efficiency, we can adopt the following strategies:
 
-1. Improve the data structure of the hash table, **allowing it to function normally in the event of a hash collision**.
-2. Only perform resizing when necessary, i.e., when hash collisions are severe.
+1. Improve the hash table data structure so that, **the hash table can work normally when hash collisions occur**.
+2. Only perform resizing when necessary, i.e., when hash collisions are relatively severe.
 
 There are mainly two methods for improving the structure of hash tables: "Separate Chaining" and "Open Addressing".
 
 ## Separate Chaining
 
-In the original hash table, each bucket can store only one key-value pair. "Separate chaining" transforms individual elements into a linked list, with key-value pairs as list nodes, storing all colliding key-value pairs in the same list. The figure below shows an example of a hash table with separate chaining.
+In the original hash table, each bucket can store only one key-value pair. "Separate chaining" converts a single element into a linked list, treating key-value pairs as list nodes, storing all colliding key-value pairs in the same linked list. The figure below shows an example of a hash table with separate chaining.
 
 ![Separate Chaining Hash Table](hash_collision.assets/hash_table_chaining.png)
 
 The operations of a hash table implemented with separate chaining have changed as follows:
 
-- **Querying Elements**: Input `key`, pass through the hash function to obtain the bucket index, access the head node of the list, then traverse the list and compare `key` to find the target key-value pair.
+- **Querying Elements**: Input `key`, obtain the bucket index through the hash function, then access the head node of the linked list, traverse the linked list, and compare `key` to find the target key-value pair.
 - **Adding Elements**: First access the list head node via the hash function, then add the node (key-value pair) to the list.
```

Comment:
optional for more clarity: "access the head node of the linked list via the hash function, then append the node (key-value pair) to the list."